sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpDataBody.java

```diff
@@ -3,26 +3,56 @@
 
 package com.azure.core.amqp.models;
 
+import com.azure.core.experimental.util.BinaryData;
 import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
 
+import java.util.Collections;
 import java.util.Objects;
 
 /**
- * This is amqp message body which represents {@link AmqpBodyType#DATA} type.
+ * This represent amqp payload of {@link AmqpBodyType#DATA} type.
  */
 public final class AmqpDataBody implements AmqpMessageBody {
-    private final IterableStream<byte[]> data;
+    private final ClientLogger logger = new ClientLogger(AmqpDataBody.class);
+    private final BinaryData binaryData;
 
     /**
-     * Creates instance of {@link AmqpDataBody} with given {@link Iterable} of byte array.
+     * Creates instance of {@link AmqpDataBody} with given {@link Iterable} of {@link BinaryData}. As of now, We
+     * support only one element in the given collection.
      *
      * @param data to be set on amqp body.
      *
      * @throws NullPointerException if {@code data} is null.
+     * @throws IllegalArgumentException if size of 'data' is zero or greater than one.
      */
-    public AmqpDataBody(Iterable<byte[]> data) {
+    public AmqpDataBody(Iterable<BinaryData> data) {
         Objects.requireNonNull(data, "'data' cannot be null.");
-        this.data = new IterableStream<>(data);
+        BinaryData payload = null;
+        for (BinaryData binaryData : data) {
+            if (payload != null) {
+                throw logger.logExceptionAsError(new IllegalArgumentException(
```

Comment:
What's the reason for only allowing a single element in the iterable?

Reply:
The underline library for AMQP we use only allow one element. But the AMQP spec allow for multiple elements.

Reply:
This appears to be a functionally breaking change from the previous code

Reply:
Should we instead make a new constructor overload `AmqpDataBody(BinaryData)` so we don't need to make a breaking change on the construct and not introduce this functional breaking change